### DISCUSSION TITLE
Dags changes are not getting reflected

### DISCUSSION THREAD

**Discussion by gigingeorge** [apache/airflow#34129] [Q&A]
I'm having airflow with local executor. The total dags re around 350. Once the dag count is increased approx 200, the dag import error, or the changes made in the dag code is not getting reflected..

Could anyone please provide assistance?

Thanks
Gigin George

**Comment by potiuk** [comment]
Look at your logs (especially dag file processor) and any warnings/errors you might see there. Also you might want to make sure to read and apply best practcies https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#top-level-python-code about top-level python code - i.e. make sure that you do not do any heavy operations when you parse DAGs (for example when you use variables in your top-level code of DAGs). 

Also  - if you use Variables at top level, in Airflow 2.7.0 there is a new, experimental feature that might help to use cache in case you are sharing a lot of variables between your DAGs. https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#use-cache

Finally you can add more schedulers if your DAG parsing becomes really long or use "Standalone DAG file processor" for your DAG parsing and have separate DAG file processor for separate sub-folders of your DAGs. 

https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#dag-processor

**Comment by gigingeorge** [comment]
But dag itself won't show in ui . If we run db upgrade manually in server, then it will come.. even if there is error in dag, that also won't show in UI

**Reply by potiuk** [reply]
Did you follow the advices in my previous answer ? What do the logs show? Are you violating best practices? Have you checked any of that? If you don't then there is no way anyone can help you .... You have to dig deeper.

**Comment by gigingeorge** [comment]
There is no error os showing..
Everything will work if the total dag count is less than 150. But i need to have more than 300..

I have followed your checklist as well.. am clueless now

**Reply by potiuk** [reply]
Without any more information everyone is even more clueless because they do not see your system. 

Check your resource usages (I am sure you must have some monitoring setup to monitor your apps) - do they seem to show some lack of resources? Memory, CPU, Disk, Network sockets opened?  I/O hiting limits?  Those are the things that you always should check whatever application you have - the resources you have are usually limiting any app and you might simply have not enouhg of them. try to use bigger machine, increase disk space, CPU, Memory if you do not have currently enough monitoring to monitor your application everything must be based on wild guessing.

Did you look at adding more schedulers as I suggested? You can also follow https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/scheduler.html#fine-tuning-your-scheduler-performance - there are some parameters like [file_parsing_sort_mode](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#config-scheduler-file-parsing-sort-mode) or [dag_dir_list_interval](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#config-scheduler-dag-dir-list-interval) that you can look at at at experiment with (and others that are described there).  Generally if you are unable to get a clue what's going on with your system with monitoring, the only way you can try do something is to experiment and make some intelligent guesses (this is wha the document allows you to do - to read, understand what knobs you can turn and intelligently guess what your problem might be and which knobs you should turn and how.

But you are the only person in the world who can do anything with it - unless you will be able to extract any useful information from logs and monitoring, I am afraid.
